contracts/Aavegotchi/facets/ItemsRolesRegistryFacet.sol

```diff
@@ -0,0 +1,408 @@
+            i--;
+            uint256 _gotchiId = _depositInfo.equippedGotchis.at(i);
+            _unequipDelegatedWearable(_gotchiId, _tokenIdToUnequip, _depositId);
+            // we remove the gotchi from the array, because enumerable set cannot be deleted (it would break the indexes)
+            _depositInfo.equippedGotchis.remove(_gotchiId);
+        }
+        // since we unequipped all items, we can reset the balance used to 0
+        delete _depositInfo.balanceUsed;
+    }
+
+    /**
+     * @notice Unequips a delegated wearable from a gotchi.
+     * @dev If The deposit is not equipped in the gotchi, it will do nothing.
+     * @param _gotchiId The gotchi to unequip the item from
+     * @param _tokenIdToUnequip The tokenId of the item to unequip
+     * @param _depositId The deposit identifier
+     */
+    function _unequipDelegatedWearable(uint256 _gotchiId, uint256 _tokenIdToUnequip, uint256 _depositId) internal {
+        GotchiEquippedDepositsInfo storage _gotchiInfo = s.gotchiEquippedDepositsInfo[_gotchiId];
+        Aavegotchi storage _aavegotchi = s.aavegotchis[_gotchiId];
+        uint256 _unequippedBalance;
+        for (uint256 slot; slot < EQUIPPED_WEARABLE_SLOTS; slot++) {
+            // if the item is not equipped in the slot or the deposit is not the same, continue
+            if (_aavegotchi.equippedWearables[slot] != _tokenIdToUnequip || _gotchiInfo.equippedDepositIds[slot] != _depositId) continue;
+
+            // if the item is equipped in the slot and the deposit is the same, unequip it by setting the slot to 0 in both arrays
+            delete _aavegotchi.equippedWearables[slot];
+            delete _gotchiInfo.equippedDepositIds[slot];
+            _unequippedBalance++;
+        }
+        // if no item was unequipped, return
+        if (_unequippedBalance == 0) return;
+        LibItems.removeFromParent(address(this), _gotchiId, _tokenIdToUnequip, _unequippedBalance);
+        emit LibERC1155.TransferFromParent(address(this), _gotchiId, _tokenIdToUnequip, _unequippedBalance);
+        // update the gotchi's equipped balance to reflect the unequipped items
+        _gotchiInfo.equippedDelegatedWearablesCount -= _unequippedBalance;
+    }
+
+    /**
+     * @notice Transfers tokens from one address to another.
+     * @dev It emits a TransferSingle event.
+     * @param _from The address to transfer from.
+     * @param _to The address to transfer to.
+     * @param _tokenAddress The token address.
+     * @param _tokenId The token identifier.
+     * @param _tokenAmount The token amount.
+     */
+    function _transferFrom(address _from, address _to, address _tokenAddress, uint256 _tokenId, uint256 _tokenAmount) internal {
+        LibItems.removeFromOwner(_from, _tokenId, _tokenAmount);
+        LibItems.addToOwner(_to, _tokenId, _tokenAmount);
+        IEventHandlerFacet(_tokenAddress).emitTransferSingleEvent(LibMeta.msgSender(), _from, _to, _tokenId, _tokenAmount);
```

Comment:
Worth noting that only the Wearables Diamond implements this function, it's not a standard erc1155 function

Reply:
Yeah, that makes total sense. The commitTokens function has a modifier that allows only Wearables NFTs to use the ERC-7589 interface, guaranteeing that _transferFrom will only be called with Wearables. If, in the future, we decide to support more ERC-1155 NFTs, we will have to modify this function, but I don't think it will be productive to change this now.